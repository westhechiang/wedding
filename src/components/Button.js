import React from 'react';
import PropTypes from 'prop-types';
import {
  TextLink,
} from 'www-coreui';

const Button = ({
  children,
  href,
}) => {
  return (
    <TextLink
      w={[300, 300, 400]}
      bg="transparent"
      color="pink.0"
      transition="0.5s ease-in-out background-color, 0.5s ease-in-out color"
      textAlign="center"
      hoverBackgroundColor="#F6D7D6"
      hoverColor="white"
      hoverBorderColor="#F6D7D6"
      border={1}
      borderColor="white"
      borderRadius={3}
      p={[10, 10, 20]}
      fontSize={24}
      fontWeight={400}
      href={href}
      target="_blank"
    >
      {children}
    </TextLink>
  );
};

Button.propTypes = {
  children: PropTypes.any,
  href: PropTypes.string,
};

export default Button;
