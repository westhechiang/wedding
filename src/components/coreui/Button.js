import styled from 'styled-components';
import { theme } from '../../ui';

const green = `
  background: ${theme.color.green};
  color: ${theme.color.white};
`;

const whiteBlue = `
  color: ${theme.color.blue4};
  background: ${theme.color.white};
  font-weight: ${theme.font.weight.semiBold};
  height: 60px;
  letter-spacing: 1px;
  line-height: 18px;
  font-size: ${theme.font.size.large};

  &:disabled {
    opacity: 0.3;
  }
`;

const whiteGray = `
  color: ${theme.color.grayLight};
  border: ${theme.border(theme.color.grayExtraLight)};
  background: ${theme.color.white};
`;

const dealDetails = `
  color: ${theme.color.blueExtraExtraLight};
  border: ${theme.border(theme.color.white)};
  background: ${theme.color.white};
  text-align: left;
  font-weight: ${theme.font.weight.medium};
`;

const blue = `
  background: ${theme.color.blueLight};
  color: ${theme.color.white};
  font-weight: ${theme.font.weight.medium};
  height: 30px;
  line-height: 30px;
  padding: 0 10px;
  font-size: ${theme.font.xxsmall};
`;

const gray = `
  background: ${theme.color.grayDark};
  background: linear-gradient(
    ${theme.color.gray} 0,
    ${theme.color.grayDark} 100%
  );
  color: ${theme.color.white};
  box-shadow: ${theme.shadow};
  font-weight: ${theme.font.weight.semiBold};
  height: 30px;
  line-height: 30px;
  padding: 0 10px;
  font-size: ${theme.font.xxsmall};
`;

const grayCondensed = `
  background: ${theme.color.grayDark};
  background: linear-gradient(
    ${theme.color.gray} 0,
    ${theme.color.grayDark} 100%
  );
  color: ${theme.color.white};
  box-shadow: ${theme.shadow};
  font-weight: ${theme.font.weight.medium};
  height: 30px;
  line-height: 30px;
  padding: 0 5px;
  font-size: ${theme.font.xxsmall};
`;

const tableFooterAddRow = `
  height: 30px;
  padding: 5px;
  line-height: 16px;
  box-sizing: border-box;
  background-color: ${theme.color.grayExtraExtraLight};
  color: ${theme.color.gray2};
  font-weight: ${theme.font.weight.semiBold};
  text-transform: uppercase;
  letter-spacing: 0.6;
  font-size: ${theme.font.xxsmall};
`;

const themes = {
  green,
  whiteBlue,
  whiteGray,
  dealDetails,
  blue,
  gray,
  grayCondensed,
  tableFooterAddRow,
};

const Button = styled.button`
  border-radius: ${theme.borderRadius};
  height: 40px;
  width: ${props => props.width || 'auto'};
  text-align: center;
  font-size: ${theme.font.small};
  font-family: ${theme.font.family};
  line-height: 40px;
  top: ${props => props.top || 'auto'};
  right: ${props => props.right || 'auto'};
  left: ${props => props.left || 'auto'};
  bottom: ${props => props.bottom || 'auto'};
  padding: 0 ${theme.gutter};
  text-transform: ${props => (props.uppercase ? 'uppercase' : 'none')};
  border: none;
  z-index: ${props => props.zIndex || 'auto'};
  position: ${props => props.position || 'static'};
  font-weight: ${theme.font.weight.light};
  transition: 0.2s ease-in-out opacity;
  ${props => themes[props.theme]};

  &:hover {
    cursor: pointer;
  }

  &:disabled {
    cursor: not-allowed;
  }
`;

export default Button;
