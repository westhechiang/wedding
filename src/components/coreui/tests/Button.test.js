import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../../../components/coreui/Button';

describe('<Button />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<Button />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "green" theme snapshot', () => {
    const wrapper = shallow(<Button theme="green" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "whiteBlue" theme snapshot', () => {
    const wrapper = shallow(<Button theme="whiteBlue" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "whiteGray" theme snapshot', () => {
    const wrapper = shallow(<Button theme="whiteGray" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "blue" theme snapshot', () => {
    const wrapper = shallow(<Button theme="blue" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "gray" theme snapshot', () => {
    const wrapper = shallow(<Button theme="gray" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "grayCondensed" theme snapshot', () => {
    const wrapper = shallow(<Button theme="grayCondensed" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
