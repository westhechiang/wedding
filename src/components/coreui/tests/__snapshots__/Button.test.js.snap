// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match its "blue" theme snapshot 1`] = `
<button
  className="sc-bdVaJa hNxhfV"
/>
`;

exports[`<Button /> should match its "gray" theme snapshot 1`] = `
<button
  className="sc-bdVaJa bnyzNl"
/>
`;

exports[`<Button /> should match its "grayCondensed" theme snapshot 1`] = `
<button
  className="sc-bdVaJa hZiCOk"
/>
`;

exports[`<Button /> should match its "green" theme snapshot 1`] = `
<button
  className="sc-bdVaJa kbTvbY"
/>
`;

exports[`<Button /> should match its "whiteBlue" theme snapshot 1`] = `
<button
  className="sc-bdVaJa bqdzO"
/>
`;

exports[`<Button /> should match its "whiteGray" theme snapshot 1`] = `
<button
  className="sc-bdVaJa bVFeIM"
/>
`;

exports[`<Button /> should match its empty snapshot 1`] = `
<button
  className="sc-bdVaJa edtxyx"
/>
`;
